internal/clock/clock.go

```diff
@@ -64,6 +64,8 @@ func (m *Mock) Add(d time.Duration) {
 
 // Timer produces a timer that will emit a time some duration after now.
 func (m *Mock) Timer(d time.Duration) *Timer {
+	m.Lock()
+	defer m.Unlock()
```

Comment:
Need to move this from lines 79 to avoid a data race on m.now when creating timers.